experimental/overhead_matching/learned/model/clevr_tokenizer.py

```diff
@@ -0,0 +1,65 @@
+from functools import reduce
+
+import common.torch as torch
+
+
+def create_tokens(batch, vocabulary):
+    """
+    A batch is a list of scenes
+    A scene is a list of items
+
+    A vocabulary is a dictionary from strings to list of strings
+    where the keys correspond to keys of an item and vocabulary[key]
+    corresponds to all possible values that the key can take.
+
+    This function assigns a unique token id for each object in the scene
+
+    This function also returns a mask where elements that are true correspond
+    to tokens that should be ignored
+    """
+
+    vocab_keys = sorted(vocabulary)
+    num_scenes = len(batch)
+    max_tokens_per_scene = reduce(max, [len(x) for x in batch])
```

Comment:
Is this more efficient than `max([len(x) for x in batch])`?

Reply:
It's not, as far as I'm aware. I was under the impression that max returned the max of two arguments, but there is also a version that returns the max of an iterable.

Good call.